controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -96,9 +96,15 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
   CONTROLLER_INTERFACE_PUBLIC
   virtual return_type update(const rclcpp::Time & time, const rclcpp::Duration & period) = 0;
 
+  CONTROLLER_INTERFACE_PUBLIC
+  void set_node(const std::shared_ptr<rclcpp_lifecycle::LifecycleNode> & node);
```

Comment:
This method is bugging me somewhat. I was thinking if we could remove it completely. 

What if we change a bit signature of `init` method. So, I propose the following changes @jackcenter, can you test if this is working? @bmagyar, @livanov93 what do you think about this?

### new `init` signature:
```
virtual return_type init(
    const std::string & controller_name, const std::string & namespace_ = "", const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions().allow_undeclared_parameters(true).automatically_declare_parameters_from_overrides(true));
```

### changing method creation in `init` method
```
node_ = std::make_shared<rclcpp_lifecycle::LifecycleNode>(controller_name, namespace_, options, false);
```

Changing starting of controller with options by calling from their `init`:
```
rclcpp::NodeOptions options;
options.allow_undeclared_parameters(true).automatically_declare_parameters_from_overrides(true);

ControllerInterface::init(controller_name, namespace_, options);
```

Reply:
I attempted to implement this, but it looks like the controller_manager tests didn't like what I did. It's not obvious to me what's wrong there, so I'll wait to see what comes of this discussion before troubleshooting further.

Reply:
Do you have somewhere updated code, so I can also take a look? Which commit should I test?

Reply:
@jackcenter I run the tests and it seems everything OK! The test that is failing is having other issues. It is not related to this PR.

Reply:
Is there anything else you would like to fix before the next review?

Reply:
Great! I thinks it's ready for another review then!